content/doc/developer/internationalization/new-internationalization.adoc

```diff
@@ -0,0 +1,339 @@
+`+src/main/resources+` with an empty value. If the file already exists,
+it will append missing entries to existing files.
+
+You then need to work on each such property file and translate messages.
+You don't have to translate the entire file — if you leave some entries
+empty, they'll fall back to the default locale.
+
+[cols="",]
+|===
+|https://addons.mozilla.org/en-US/firefox/addon/1333[Quick locale
+switcher] firefox extension is useful to toggle between various locales.
+|===
+
+=== Translating static HTML resources
+
+Stand-alone HTML files are often used in Jenkins for things like inline
+help messages. These resources need to be translated by adding the
+locale code between the file name and the extension. For example, the
+Japanese version of `+abc.html+` would be `+abc_ja.html+`, and British
+version of it could be `+abc_en_GB.html+`. These files need to be
+encoded in UTF-8.
+
+=== Pushing changes
+
+Once you made some changes, you can commit them. Translators should
+consider themselves as owning property files for their locale, so feel
+free to go ahead and just commit. If you are new to this, doing a small
+commit first is a good idea. You can also always send in a patch if you
+prefer to be safe.
+
+When starting a translation, try to check if there's anyone else working
+on the same locale. You can find out who they are by finding existing
+localization and looking at its history. Try to get in touch with them
+to avoid a surprise.
+
+== Stapler plugin for IntelliJ IDEA
+
+Refactoring the existing code to handle i18n correctly is tedious. So
+https://wiki.jenkins.io/display/JENKINS/IntelliJ+IDEA+plugin+for+Stapler[IntelliJ
+IDEA plugin for Stapler] is developed to simplify this (note that
+JetBrains kindly offered the open-source license for the Jenkins
+project, so contributers can get the license for free — contact Kohsuke
+if you need one.)
+
+Once installed, this adds a menu item in the main menu, under
+"Refactor." It is highly recommended to give some keyboard short cut to
+this. I use "Ctrl+Alt+I": +
+image:../images/developer/internationalization/menuItem.png[image,title="Jenkins > Internationalization > menuItem.png"] +
+Now, to use this feature, select the message you'd like to
+internationalize, then trigger this refactoring command. It'll ask you
+the message property name for this, so pick a name: +
+image:../images/developer/internationalization/before.png[image,title="Jenkins > Internationalization > before.png"] +
+IDEA will place the message into the resource file, and adjust the code
+accordingly. Note that an error is highlighted until you run
+`+mvn compile+` again to generate new methods on the `+Messages+`
+class. +
+image:../images/developer/internationalization/after.png[image,title="Jenkins > Internationalization > after.png"]
+
+== Stapler plugin for NetBeans
+
+See
+https://wiki.jenkins.io/display/JENKINS/NetBeans+plugin+for+Stapler[NetBeans
+plugin for Stapler] for details.
```

Comment:
Another broken link.  This is also discussed in https://www.jenkins.io/doc/developer/development-environment/ide-configuration/ , with a link to http://plugins.netbeans.org/plugin/43938/?show=true , but the netbeans link also seems to be broken.  The github repo for this plugin is https://github.com/stapler/netbeans-stapler-plugin -- this is what is referenced from the https://www.jenkins.io/doc/developer/tutorial/prepare/ page.

Reply:
No much more content in the original wiki page alas:
> Support for Stapler and other Jenkins core/plugin development needs from the NetBeans IDE.
> [GitHub home](https://github.com/stapler/netbeans-stapler-plugin)
> [Plugin Portal downloads](http://plugins.netbeans.org/plugin/43938/?show=true)